crates/anvil/src/eth/api.rs

```diff
@@ -2230,84 +2230,43 @@ impl EthApi {
 
         // execute the call without writing to db
         let ethres =
             self.backend.call_with_state(&state, call_to_estimate, fees.clone(), block_env.clone());
 
-        // Exceptional case: init used too much gas, we need to increase the gas limit and try
-        // again
-        if let Err(BlockchainError::InvalidTransaction(InvalidTransactionError::GasTooHigh(_))) =
-            ethres
-        {
-            // if price or limit was included in the request then we can execute the request
-            // again with the block's gas limit to check if revert is gas related or not
-            if request.gas.is_some() || request.gas_price.is_some() {
-                return Err(map_out_of_gas_err(
-                    request,
-                    state,
-                    self.backend.clone(),
-                    block_env,
-                    fees,
-                    gas_limit,
-                ));
+        let gas_used = match ethres.try_into()? {
+            GasEstimationCallResult::Success(gas) => Ok(U256::from(gas)),
+            GasEstimationCallResult::OutOfGas => {
+                Err(InvalidTransactionError::BasicOutOfGas(highest_gas_limit).into())
             }
-        }
-
-        let (exit, out, gas, _) = ethres?;
-        match exit {
-            return_ok!() => {
-                // succeeded
-            }
-            InstructionResult::OutOfGas | InstructionResult::OutOfFunds => {
-                return Err(InvalidTransactionError::BasicOutOfGas(gas_limit).into())
-            }
-            // need to check if the revert was due to lack of gas or unrelated reason
-            // we're also checking for InvalidFEOpcode here because this can be used to trigger an error <https://github.com/foundry-rs/foundry/issues/6138> common usage in openzeppelin <https://github.com/OpenZeppelin/openzeppelin-contracts/blob/94697be8a3f0dfcd95dfb13ffbd39b5973f5c65d/contracts/metatx/ERC2771Forwarder.sol#L360-L367>
-            return_revert!() | InstructionResult::InvalidFEOpcode => {
-                // if price or limit was included in the request then we can execute the request
-                // again with the max gas limit to check if revert is gas related or not
-                return if request.gas.is_some() || request.gas_price.is_some() {
-                    Err(map_out_of_gas_err(
-                        request,
-                        state,
-                        self.backend.clone(),
-                        block_env,
-                        fees,
-                        gas_limit,
-                    ))
-                } else {
-                    // the transaction did fail due to lack of gas from the user
-                    Err(InvalidTransactionError::Revert(Some(convert_transact_out(&out).0.into()))
-                        .into())
-                };
+            GasEstimationCallResult::Revert(output) => {
+                Err(InvalidTransactionError::Revert(output).into())
```

Comment:
I think we want to keep mapping the output here: `convert_transact_out`?

Reply:
We handle conversion to bytes in `TryFrom` impl and it's basically the only thing `convert_transact_out` is doing